algorithms/ar-expsq/c/exponentiation_by_squaring.c

```diff
@@ -0,0 +1,24 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+void exponentiationSquaring(long base, long power) {
+	long result = 1;
+
+	while(power > 0) { //run loop till power is greater than zero
+		if(power % 2 != 0) { //if power is ODD multiply result by base
+			result = result * base;
+		}
+		base = base * base; //squaring base
+		power /= 2; //reduce power to half
+	}
+	
+	printf("%lu\n", result);
```

Comment:
Codacy found an issue: [%lu in format string (no. 1) requires 'unsigned long' but the argument type is 'signed long'.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6357663)